src/dmd/statementsem.d

```diff
@@ -3039,7 +3047,24 @@ else
 
             // for static alias this: https://issues.dlang.org/show_bug.cgi?id=17684
             if (rs.exp.op == TOK.type)
-                rs.exp = resolveAliasThis(sc, rs.exp);
+            {
+                Expression[] results;
+                Expression e = rs.exp.copy();
```

Comment:
And finally here (may have missed a few). All noted examples are where the caller is having to handle alias this, but it really shouldn't.